lambdas/process-cri-callback/src/test/java/uk/gov/di/ipv/core/processcricallback/pact/cicCri/ContractTest.java

```diff
@@ -0,0 +1,503 @@
+                                assertEquals("Mary", nameParts.get(0).get("value").asText());
+                                assertEquals("Watson", nameParts.get(1).get("value").asText());
+                            } catch (VerifiableCredentialException
+                                    | ParseException
+                                    | JsonProcessingException e) {
+                                throw new RuntimeException(e);
+                            }
+                        });
+    }
+
+    @Pact(provider = "CicCriProvider", consumer = "IpvCoreBack")
+    public RequestResponsePact inValidRequestReturns400(PactDslWithProvider builder)
+            throws Exception {
+        return builder.given("dummyApiKey is a valid api key")
+                .given("dummyAccessToken is a valid access token")
+                .given("test-subject is a valid subject")
+                .given("dummyClaimedIdentityComponentId is a valid issuer")
+                .given("VC givenName is Mary")
+                .given("VC familyName is Watson")
+                .uponReceiving("Valid credential request for VC")
+                .path("/credential")
+                .method("POST")
+                .headers("x-api-key", PRIVATE_API_KEY, "Authorization", "Bearer dummyAccessToken")
+                .willRespondWith()
+                .status(400)
+                .body(
+                        new PactJwtIgnoreSignatureBodyBuilder(
+                                VALID_VC_HEADER, INVALID_VC_BODY, IN_VALID_VC_SIGNATURE))
+                .toPact();
+    }
+
+    @Test
+    @PactTestFor(pactMethod = "inValidRequestReturns400")
+    void fetchVerifiableCredential_whenCalledAgainstCicCri_retrievesAnInValidVc(
+            MockServer mockServer) throws URISyntaxException, CriApiException {
+        // Arrange
+        var credentialIssuerConfig = getMockCredentialIssuerConfig(mockServer);
+        configureMockConfigService(credentialIssuerConfig);
+
+        // We need to generate a fixed request, so we set the secure token and expiry to constant
+        // values.
+        var underTest =
+                new CriApiService(
+                        mockConfigService, mockSigner, mockSecureTokenHelper, CURRENT_TIME);
+
+        // Act
+
+        CriApiException exception =
+                assertThrows(
+                        CriApiException.class,
+                        () ->
+                                underTest.fetchVerifiableCredential(
+                                        new BearerAccessToken("dummyAccessToken"),
+                                        getCallbackRequest("dummyAuthCode", credentialIssuerConfig),
+                                        getCriOAuthSessionItem()));
+        // Assert
+        assertEquals(
+                "Failed to get credential from issuer", exception.getErrorResponse().getMessage());
+        assertEquals(500, exception.getHttpStatusCode());
```

Comment:
I see, but I don't think we really expect a scenario where this endpoint returns us a 400 with an 'invalid' VC? the existing contract tests seem to just be checking the invalid access token scenario (which I think we're missing here?)

Reply:
Should the CRI be able to send us a VC without a birthday?
- If they can then I guess it's not an invalid VC and we should have a test to cover it (but why would they also send a 400 return code if it's valid?)
- If they can't then we're testing something that should never happen - how would the CIC team write a test to generate a broken VC if their code doesn't allow it?

As Patrick says, if they do send us a 400 response (bad request) then they probably won't send us a VC at all because they couldn't understand the request.
Why do you expect them to send a 400 in response to the request in the pact (which looks fine)?

Reply:
Thanks @shivanshuit914. I think we're still missing a test case we do want, for invalid access token on the `/credential` endpoint - something like this: https://github.com/govuk-one-login/ipv-core-back/blob/53c001d5fbe63dde0f385389148480e41a99743b/lambdas/process-cri-callback/src/test/java/uk/gov/di/ipv/core/processcricallback/pact/passportCri/ContractTest.java#L326 ?